internal/controller/controller.go

```diff
@@ -223,36 +223,52 @@ func (c *Controller) startInformers(stopCh <-chan struct{}) {
 }
 
 func (c *Controller) createMeshServices() error {
-	services, err := c.clients.GetServices(metav1.NamespaceAll)
+	potentialSvcs, err := c.clients.GetServicesWithSelectors(
+		metav1.NamespaceAll,
+		c.ignored.LabelSelector(),
+		c.ignored.FieldSelector(),
+	)
 	if err != nil {
 		return fmt.Errorf("unable to get services: %v", err)
 	}
 
-	for _, service := range services {
-		if c.ignored.IsIgnoredService(service.Name, service.Namespace) {
-			continue
-		}
+	maeshSvcs, err := c.clients.GetServicesWithSelectors(c.meshNamespace, "app==maesh", "")
+	if err != nil {
+		return fmt.Errorf("unable to get maesh services: %v", err)
+	}
 
+	maeshSvcsByName := reduceServicesByName(maeshSvcs)
+
+	for _, service := range potentialSvcs {
 		log.Debugf("Creating mesh for service: %v", service.Name)
+
 		meshServiceName := c.userServiceToMeshServiceName(service.Name, service.Namespace)
 
-		for _, subservice := range services {
-			// If there is already a mesh service created, don't bother recreating
-			if subservice.Name == meshServiceName && subservice.Namespace == c.meshNamespace {
-				continue
-			}
+		if _, found := maeshSvcsByName[meshServiceName]; found {
+			log.Debugf("Skipping already created maesh service: %s", service.Name)
```

Comment:
Not sure if this log is needed. It can really really spam the log in larger clusters.